README.md

```diff
@@ -42,11 +42,25 @@ go get github.com/oklog/ulid/v2
 An ULID is constructed with a `time.Time` and an `io.Reader` entropy source.
 This design allows for greater flexibility in choosing your trade-offs.
 
-Please note that `rand.Rand` from the `math` package is *not* safe for concurrent use.
-Instantiate one per long living go-routine or use a `sync.Pool` if you want to avoid the potential contention of a locked `rand.Source` as its been frequently observed in the package level functions.
+If you want sane defaults, just use `ulid.Make()` which will produce a per process monotonically increasing ULID based on the current time that is safe for concurrent use.
 
 ```go
-func ExampleULID() {
+func ExampleMake() {
+	fmt.Println(ulid.Make())
+	// Output: 0000XSNJG0MQJHBF4QX1EFD6Y3
+}
+```
+
+Otherwise, you'll need to provide your own entropy source. Since `rand.Rand` from the `math` package is [*not* safe for concurrent use](https://github.com/golang/go/issues/3611), consider the following guidance:
+
+- Instantiate one `rand.Rand` per long living go-routine if your concurrency model permits — this option will result in no lock contention.
+
+- For usage in short lived go-routines (e.g. HTTP handlers), use [`golang.org/x/exp/rand.Rand`](https://pkg.go.dev/golang.org/x/exp/rand#example-LockedSource) which is safe for concurrent use, but can result in a high amount of lock contention if many go-routines concurrently call `Read` on it.
+
+- When lock contention is too high with the previous approach, use a `sync.Pool` of `rand.Rand` instances. This won't provide any benefits if you need to per process monotonic ULIDs, since using `LockedMonotonicReader` will synchronize all reads anyways.
+
```

Comment:
```markdown
ULIDs are constructed from two things: a timestamp with millisecond precision,
and some random data.

Timestamps are modeled as uint64 values representing a Unix time in milliseconds. 
They can be produced by passing a [time.Time](https://pkg.go.dev/time#Time) to
[ulid.Timestamp](https://pkg.go.dev/github.com/oklog/ulid/v2#Timestamp), 
or by calling  [time.Time.UnixMilli](https://pkg.go.dev/time#Time.UnixMilli) 
and converting the returned value to `uint64`.

Random data is taken from a provided [io.Reader](https://pkg.go.dev/io#Reader).
This design allows for greater flexibility when choosing trade-offs, but can be
a bit confusing to newcomers.

If you just want to generate a ULID and don't (yet) care about details like
performance, cryptographic security, monotonicity, etc., use the
[ulid.Make](https://pkg.go.dev/github.com/oklog/ulid/v2#Make) helper function.
This function calls [time.Now](https://pkg.go.dev/time#Now) to get a timestamp,
and uses a process-global, pseudo-random (i.e. based on
[math/rand](https://pkg.go.dev/math/rand)), and monotonic (i.e. wrapped by
[ulid.MonotonicReader](https://pkg.go.dev/oklog/ulid/v2#MonotonicReader)) source
of entropy.

Security-sensitive use cases should always use
[crypto/rand](https://pkg.go.dev/crypto/rand) instead of
[math/rand](https://pkg.go.dev/math/rand).

Performance-sensitive use cases should avoid synchronization when generating
entropy. But be careful: [math/rand.Rand](https://pkg.go.dev/math/rand#Rand) is
not safe for concurrent use. One option is to use a unique source of entropy for
each concurrent goroutine, which results in no lock contention, but cannot
provide strong guarantees about the random data, and does not provide
monotonicity within a given millisecond. 

One common performance optimization is to pool sources of entropy using a
[sync.Pool](https://pkg.go.dev/sync#Pool).

Monotonicity is a property that says each ULID is "bigger than" the previous
one. If you generate ULIDs at a rate of one per millisecond or slower, they are
automatically monotonic, because ULIDs have a timestamp field with millisecond
precision. If you generate ULIDs at a rate greater than one per millisecond, the
ULIDs generated in a given millisecond share the same timestamp field, and are
ordered by their random component. By default, ULIDs that share the same
millisecond timestamp are un-ordered. You can use
[ulid.MonotonicEntropy](https://pkg.go.dev/oklog/ulid/v2#MonotonicEntropy) to
create ULIDs which are monotonic within a given millisecond, with caveats. See
the documentation for details.
```

Reply:
Probably other pitfalls that should be called out here. Something about not calling `time.Now` in a hot loop? More?

Reply:
Maybe also example code for each paragraph?